### PR TITLE
Add connection_pool

### PR DESCRIPTION
**Pull request by miharekar**
We encountered an error when using thrift in a multi-threaded environment (sidekiq).

This PR uses connection_pool which provides a pool for all the threads to use.

### CODE REVIEW

File: lib/twofishes.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 require 'twofishes/version'
 require 'twofishes/configuration'
-require 'thrift_client'
+require 'twofishes/twofishes_client'
```
Comment (masone) [inline lib/twofishes.rb]: isn't that `twofishes/client`?

File: lib/twofishes/twofishes_client.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require 'connection_pool'
+require 'thrift_client'
+
+$twofishes_client = ConnectionPool::Wrapper.new(timeout: Twofishes.configuration.timeout) {
```
Comment (masone) [inline lib/twofishes/twofishes_client.rb]: should we make `size` configurable?

File: lib/twofishes.rb (hunk 1 above)
Comment (miharekar) [inline lib/twofishes.rb]: It's the new file which holds the connection pool. Better idea how to name it?

File: lib/twofishes/twofishes_client.rb (hunk 2 above)
Comment (miharekar) [inline lib/twofishes/twofishes_client.rb]: This is the reason we're doing this - because we can't have more than 1.

**Pull request closed without merging**